lib/usb/usb_standard.c

```diff
@@ -129,8 +129,9 @@ static uint16_t build_config_descriptor(usbd_device *usbd_dev,
 		}
 	}
 
-	/* Fill in wTotalLength. */
-	*(uint16_t *)(tmpbuf + 2) = totallen;
+	/* Fill in wTotalLength.
+	 * Note that tmpbuf is sometimes not halfword-aligned */
+	memcpy((tmpbuf + 2), &totallen, sizeof(uint16_t));
```

Comment:
here again, calling memcpy for 2 bytes seems like total overkill. It will work, but....

Reply:
Often, gcc is able to optimize memcpy if the size is known. For example compiling on STM32L4:
```
void copy_uint16(void *a, const void *b) {
    memcpy(a, b, sizeof(uint16_t));
}
   0:   880b            ldrh    r3, [r1, #0]
   2:   8003            strh    r3, [r0, #0]
   4:   4770            bx      lr
```

Reply:
Interesting, I seem to recall testing something similar a while ago and gcc didn't do that optimization.